src/main/java/ch/njol/skript/doc/JSONGenerator.java

```diff
@@ -0,0 +1,247 @@
+package ch.njol.skript.doc;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.ClassInfo;
+import ch.njol.skript.lang.SkriptEventInfo;
+import ch.njol.skript.lang.SyntaxElement;
+import ch.njol.skript.lang.SyntaxElementInfo;
+import ch.njol.skript.lang.function.Functions;
+import ch.njol.skript.lang.function.JavaFunction;
+import ch.njol.skript.registrations.Classes;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import org.eclipse.jdt.annotation.Nullable;
+import org.skriptlang.skript.lang.structure.Structure;
+import org.skriptlang.skript.lang.structure.StructureInfo;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+/**
+ * Generates JSON docs
+ */
+public class JSONGenerator extends DocumentationGenerator {
+
+	public JSONGenerator(File templateDir, File outputDir) {
+		super(templateDir, outputDir);
+	}
+
+	/**
+	 * Coverts a String array to a JsonArray
+	 * @param strings the String array to convert
+	 * @return the JsonArray containing the Strings
+	 */
+	private static @Nullable JsonArray convertToJsonArray(String @Nullable[] strings) {
+		if (strings == null)
+			return null;
+		JsonArray jsonArray = new JsonArray();
+		for (String string : strings)
+			jsonArray.add(new JsonPrimitive(string));
+		return jsonArray;
+	}
+
+	/**
+	 * Generates the documentation JsonObject for an element that is annotated with documentation
+	 * annotations (e.g. effects, conditions, etc.)
+	 * @param syntaxInfo the syntax info element to generate the documentation object of
+	 * @return the JsonObject representing the documentation of the provided syntax element
+	 */
+	private @Nullable JsonObject generatedAnnotatedElement(SyntaxElementInfo<?> syntaxInfo) {
+		Class<?> syntaxClass = syntaxInfo.getElementClass();
+		Name nameAnnotation = syntaxClass.getAnnotation(Name.class);
+		if (nameAnnotation == null || syntaxClass.getAnnotation(NoDoc.class) != null)
+			return null;
+		JsonObject syntaxJsonObject = new JsonObject();
+		syntaxJsonObject.addProperty("id", DocumentationIdProvider.getId(syntaxInfo));
+		syntaxJsonObject.addProperty("name", nameAnnotation.value());
+
+		Since sinceAnnotation = syntaxClass.getAnnotation(Since.class);
```

Comment:
what if `@Since` isn't present?

Reply:
this is based off tribal knowledge from ayham. htmlgenerator treats it as non-null currently

Reply:
nevermind it totally doesnt